spec.md

```diff
@@ -388,13 +388,34 @@ both appear within the same JSON object.
   or application protocol's content-type metadata property. Normative rules
   for the binary mode and the content-type metadata mapping can be found 
   in the respective transport mapping specifications.
-    
+  
 * Constraints:
   * OPTIONAL
   * If present, MUST adhere to the format specified in
     [RFC 2046](https://tools.ietf.org/html/rfc2046)
 * For Media Type examples see [IANA Media Types](http://www.iana.org/assignments/media-types/media-types.xhtml)
 
+### correlationProperties
+* Type: `Map`
+* Description: A list of "key-value" pairs providing unique identification
+  information about the event, which can be used by the serverless 
+  platform to correlate multiple events associated with a serverless
+  application workflow. The value MUST be a string. There MUST be no
+  duplication of keys. 
+* Constraints:
+  * REQUIRED, but can be empty if all unique identification information is 
+  already provided in the source etc. attributes. 
+* Examples:
+``` JSON
+  correlationProperties
```

Comment:
I can understand the needs to put correlation in the spec but I still worry about this "open" format that puts no restriction on what can be put in this key-value map. Is there no way that we can provide pre-defined fields? If there is no way, then this does not feel like something that can be defined in a spec.